promql/parser/parse.go

```diff
@@ -685,15 +686,18 @@ func (p *parser) addOffset(e Node, offset time.Duration) {
 		offsetp = &s.Offset
 		endPosp = &s.PosRange.End
 	case *MatrixSelector:
-		if vs, ok := s.VectorSelector.(*VectorSelector); ok {
-			offsetp = &vs.Offset
+		vs, ok := s.VectorSelector.(*VectorSelector)
+		if !ok {
+			// This invalid query will be caught at some other layer.
```

Comment:
Why not catch it here?

Reply:
To avoid 2 places trying to catch same error, can lead to error message inconsistencies.

Reply:
I'd generally encourage the belts and suspenders approach, so we don't risk an error slipping through.